src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Entry<T> first;
+    private Entry<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        this.size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Entry<T> tempLast = last;
+        last = new Entry<>(value, tempLast, null);
+        if (tempLast == null) {
+            first = last;
+        } else {
+            tempLast.next = last;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of range!");
+        }
+        if (size == index) {
+            add(value);
+        } else {
+            Entry<T> tempEntry = getEntryByIndex(index);
+            if (tempEntry.previous == null) {
+                first = new Entry<>(value, null, tempEntry);
+            } else {
+                Entry<T> currentEntry = new Entry<>(value, tempEntry.previous, tempEntry);
+                tempEntry.previous.next = currentEntry;
+                tempEntry.previous = currentEntry;
+            }
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        if (list.size() == 0) {
+            return false;
+        }
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getEntryByIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Entry<T> currentEntry = getEntryByIndex(index);
+        T oldValue = currentEntry.element;
+        currentEntry.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Entry<T> removedEntry = getEntryByIndex(index);
+        if (removedEntry.previous == null) {
+            first = removedEntry.next;
+        } else {
+            removedEntry.previous.next = removedEntry.next;
+        }
+        if (removedEntry.next == null) {
+            last = removedEntry.previous;
+        } else {
+            removedEntry.next.previous = removedEntry.previous;
+        }
+        size--;
```

Comment:
let's move this code at separate method `unlink(Node<T> nodeToRemove)`  
then we can call this method in remove(T t)